File: AIO/AIO 2020/[p3] Ghost Encounters.py
N, K = map(int, input().split())
 
freq_table = dict()
current_max = 1
 
for _ in range(N):
    x, t = map(int, input().split())
    current_data = (x*K)-t
    if current_data in freq_table:
        freq_table[current_data] += 1
    else:
        freq_table[current_data] = 1
    if freq_table[current_data] > current_max:
        current_max = freq_table[current_data]
 
print(current_max)

File: AIO/AIO 2020/[p3] Ghost Encounters.cpp
#include <bits/stdc++.h>
using namespace std;
 
signed int main() {
    int N, K, x, t;
    cin >> N >> K;
    map<int, int> freq_table;
    int current_max = 1, current_data;
    for(int i = 0; i < N; i++) {
        cin >> x >> t;
        current_data = (x*K) -t;
        freq_table[current_data]++;
 
        if(freq_table[current_data] > current_max) {
            current_max = freq_table[current_data];
        }
    }
    cout << current_max;
}
